soc/espressif/esp32/default.ld

```diff
@@ -680,10 +680,27 @@ SECTIONS
 
   .dram0.noinit (NOLOAD) :
   {
-    . = ALIGN (8);
+    . = ALIGN (4);
+    __dram_noinit_start = ABSOLUTE(.);
+#ifdef CONFIG_ESP32_WIFI_NET_ALLOC_SPIRAM
+    *(EXCLUDE_FILE(
+      *libdrivers__wifi.a:*
+      *libsubsys__net__l2__ethernet.a:*
+      *libsubsys__net__lib__config.a:*
+      *libsubsys__net__ip.a:*
+      *libsubsys__net.a:* ) .noinit)
+    *(EXCLUDE_FILE(
+      *libdrivers__wifi.a:*
+      *libsubsys__net__l2__ethernet.a:*
+      *libsubsys__net__lib__config.a:*
+      *libsubsys__net__ip.a:*
+      *libsubsys__net.a:* ) .noinit.*)
+#else
     *(.noinit)
     *(.noinit.*)
-    . = ALIGN (8);
+#endif /* CONFIG_ESP32_WIFI_NET_ALLOC_SPIRAM */
+    __dram_noinit_end = ABSOLUTE(.);
+    . = ALIGN (4);
```

Comment:
There is something wrong with this logic. In case `CONFIG_ESP32_WIFI_NET_ALLOC_SPIRAM` is defined, it should only exclude the `net/wifi` stuff. Currently, it is also not adding `.noinit` data at all. Is that right?